### PR TITLE
Updating reno usage: message template and documentation

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Updating Reno sections list in the base template and documenting them.

### CODE REVIEW

**Review by xvello — DISMISSED** [review]
👍

File: docs/dev/contributing.md
Comment (olivielpeau) [inline docs/dev/contributing.md]: `sections`

File: docs/dev/contributing.md
Comment (olivielpeau) [inline docs/dev/contributing.md]: `others`

**Review by olivielpeau — COMMENTED** [review]
Nice! 🍰 

A few inline nits + general comments:
* you also need to configure `sections` in reno's config, so that it can match our custom sections in the template to actual sections in the generated changelog
* should we mention somewhere that each section note should be formatted with reStructuredText?

File: docs/dev/contributing.md
Comment (olivielpeau) [inline docs/dev/contributing.md]: still not very clear to me what this means :/ Is this section for fixed security issues or known security issues? (if we want to have sections for both, then I think we should have 2 different sections)

**Review by olivielpeau — DISMISSED** [review]
one nit left, but LGTM!

2 more questions:
* are we getting rid of the `other` section? (we've used it quite a lot in the past, see current changelog)
* what's your plan to ship this change (since the sections have changed, I guess some of the existing release notes and some of those that are not merged yet will become "invalid" when this is merged)

waiting for a review from @DataDog/agent-integrations :)

File: docs/dev/contributing.md
Comment (hush-hush) [reply docs/dev/contributing.md]: I think it's about security related stuff: either security issue or fix, but also new behavior that may improve it ...

I think one section is fine, we almost never use it and it's more a way for user to quickly see what is important.

**Comment by hush-hush** [comment]
@olivielpeau I added the `other` section (completely forgot about it). I still added in the documentation and template that this should rarely be used as I feel a lot of the current entries should be in other section (mainly the 'enhancement' section that was missing).

For the rest my plan was to merged after the release of `6.3.0` so almost no new release note has been merged.

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**
